hasktorch/src/Torch/Functional.hs

```diff
@@ -289,6 +289,20 @@ relu
     -> Tensor -- ^ output
 relu t = unsafePerformIO $ (cast1 ATen.relu_t) t
 
+-- | Applies exponential linear unit function element wise with default alpha value = 1
+elu
+    :: Tensor -- ^ input
+    -> Tensor -- ^ output
+elu t = unsafePerformIO $ (cast1 ATen.elu_t) t
+
+-- | Applies Exponential linear unit function element-wise, with alpha input, \(\text{ELU}(x) = \max(0,x) + \min(0, \alpha * (\exp(x) - 1))\)
+elu_a
+    :: Scalar s 
+    => Tensor -- ^ input
+    -> s -- ^ alpha value for ELU formulation
+    -> Tensor -- ^ output
+elu_a t a = unsafePerformIO $ (cast2 ATen.elu_ts) t a
+
```

Comment:
For now, we use suffix of `'` with default argument.
I think it is better to put input tensor into last argument like following code.

```haskell
elu :: Scalar s => s -> Tensor -> tensor
elu' :: Tensor -> Tensor
```

Reply:
Hey, yeah. I was going through the currying conversation on slack today and realised. Will update this asap.

Reply:
@junjihashimoto Please have a look now?

Reply:
Thx!
There is no test of elu on both untyped and typed spec.
Could you add the tests of elu?

Reply:
Sure.

Reply:
@junjihashimoto This is ready for review.